modules/nf-core/mageck/count/main.nf

```diff
@@ -12,27 +12,35 @@ process MAGECK_COUNT {
     path(library)
 
     output:
-    tuple val(meta), path("*count.txt")           , emit: count
+    tuple val(meta), path("*count.txt"),             emit: count
     tuple val(meta), path("*.count_normalized.txt"), emit: norm
-    path "versions.yml"                            , emit: versions
+    tuple val(meta), path("*.countsummary.txt"),     emit: summary
+    tuple val(meta), path("*.count_normalized.txt"), emit: normalized
+    tuple val(meta), path("*.log"),                  emit: logs
+    path "versions.yml",                             emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def input_file = ("$inputfile".endsWith(".fastq.gz")) ? "--fastq ${inputfile}" : "-k ${inputfile}"
     def sample_label = ("$inputfile".endsWith(".fastq.gz") || "$inputfile".endsWith(".fq.gz")) ? "--sample-label ${meta.id}" : ''
 
+    if (meta.single_end && ("$inputfile".endsWith(".fastq.gz") || "$inputfile".endsWith(".fq.gz"))) {
```

Comment:
This seems inconsistent, what does `-k` do before? It had a `"-k ${inputfile}"` when not ending in `fastq.gz` before.